RGLServerPlugin/src/Lidar.cc

```diff
@@ -67,6 +67,17 @@ bool RGLServerPluginInstance::LoadConfiguration(const std::shared_ptr<const sdf:
     if (!LidarPatternLoader::Load(sdf, lidarPattern)) {
         return false;
     }
+
+    // Check for 2d pattern and get LaserScan parameters
+    if (sdf->HasElement("pattern_lidar2d")) {
```

Comment:
Hmm, this does sound like a good idea but I think I might have the wrong interpretation here. So to clarify, the idea is to move this `sdf->HasElement` check to the `LidarPatternLoader::Load` function and then still have a condition here in the `LoadConfiguration` function to set the respective variables/config for the message type. correct?

What I think I have interpreted wrongly (but honestly think is cool) is that we create an overloaded function for `CreateMessage` with different types and then do just universally call that function to avoid all these if elses.

I could look into the latter option but if it sounds too complicated I can go with the first one for now.

Reply:
> Hmm, this does sound like a good idea but I think I might have the wrong interpretation here. So to clarify, the idea is to move this sdf->HasElement check to the LidarPatternLoader::Load function and then still have a condition here in the LoadConfiguration function to set the respective variables/config for the message type. correct?

This is correct, with one minor clarification. I mean removing `sdf->HasElement` check and setting `PublishMessageType` in each of the `patternLoadFunctions` (`LoadPatternFromUniform`, `LoadPatternFromCustom`, `LoadPatternFromLidar2d`, ...).

> What I think I have interpreted wrongly (but honestly think is cool) is that we create an overloaded function for CreateMessage with different types and then do just universally call that function to avoid all these if else's.

I haven't thought about it, but it also seems like a good idea.

> I could look into the latter option but if it sounds too complicated I can go with the first one for now.

The first option will be completely sufficient at this point. I would also prefer not to overstay this PR.

Reply:
> The first option will be completely sufficient at this point. I would also prefer not to overstay this PR.

Yea that makes sense.